apps/rehabstod/src/store/api.ts

```diff
@@ -55,20 +55,27 @@ export const api = createApi({
     getLinks: builder.query<Record<string, Link | undefined>, void>({
       query: () => 'config/links',
     }),
-    getSickLeaves: builder.mutation<SickLeaveInfo[], void>({
-      query: () => ({
+    getSickLeaves: builder.mutation<SickLeaveInfo[], ActiveSickLeavesRequest>({
+      query: (request) => ({
         url: 'sickleaves/active',
         method: 'POST',
+        body: request,
       }),
       transformResponse: (response: { content: SickLeaveInfo[] }) => response.content,
     }),
+    getPopulatedFilters: builder.query<PopulateFiltersResponse, void>({
```

Comment:
We can invalidate the cached filter result so that filter gets called again if the user change unit.

Add `providesTags: ['SickLeavesFilter'],` to `getPopulatedFilters` and `invalidatesTags: ['SickLeavesFilter'],` to `changeUnit`. `tagTypes` also needs to be updated with the new tag.